dspy/retrieve/databricks_rm.py

```diff
@@ -55,53 +47,72 @@ class DatabricksRM(dspy.Retrieve):
+                used to identify the token based on the current environment.
+            columns (list[str], optional): Extra column names to include in response,
+                in addition to the document id and text columns specified by
+                ``docs_id_column_name`` and ``text_column_name``.
+            filters_json (str, optional): A JSON string specifying additional query filters.
+                Example filters: ``{"id <": 5}`` selects records that have an ``id`` column value
+                less than 5, and ``{"id >=": 5, "id <": 10}`` selects records that have an ``id``
+                column value greater than or equal to 5 and less than 10.
+            k (int): The number of documents to retrieve.
+            docs_id_column_name (str): The name of the column in the Databricks Vector Search Index
+                containing document IDs.
+            text_column_name (str): The name of the column in the Databricks Vector Search Index
+                containing document text to retrieve.
+        """
         super().__init__(k=k)
-        if not databricks_token and not os.environ.get("DATABRICKS_TOKEN"):
-            raise ValueError(
-                "You must supply databricks_token or set environment variable DATABRICKS_TOKEN"
-            )
-        if not databricks_endpoint and not os.environ.get("DATABRICKS_HOST"):
-            raise ValueError(
-                "You must supply databricks_endpoint or set environment variable DATABRICKS_HOST"
-            )
-        if not databricks_index_name:
-            raise ValueError("You must supply vector index name")
-        if not columns:
-            raise ValueError(
-                "You must specify a list of column names to be included in the response"
-            )
-        self.databricks_token = (
-            databricks_token if databricks_token else os.environ["DATABRICKS_TOKEN"]
-        )
+        self.databricks_token = databricks_token if databricks_token is not None else os.environ.get("DATABRICKS_TOKEN")
         self.databricks_endpoint = (
-            databricks_endpoint
-            if databricks_endpoint
-            else os.environ["DATABRICKS_HOST"]
+            databricks_endpoint if databricks_endpoint is not None else os.environ.get("DATABRICKS_HOST")
         )
+        if not _databricks_sdk_installed and (self.databricks_token, self.databricks_endpoint).count(None) > 0:
+            raise ValueError(
+                "To retrieve documents with Databricks Vector Search, you must install the"
+                " databricks-sdk Python library, supply the databricks_token and"
+                " databricks_endpoint parameters, or set the DATABRICKS_TOKEN and DATABRICKS_HOST"
+                " environment variables."
+            )
         self.databricks_index_name = databricks_index_name
-        self.columns = columns
+        self.columns = list({docs_id_column_name, text_column_name, *(columns or [])})
```

Comment:
Automatically include the user-specified `docs_id_column` and `text_column` in the list of columns to retrieve from the vector search index (`self.columns`). Previously, users had to specify the docs ID and text columns in 2 places (the corresponding arg *and* the `columns` arg). If the user forgot to specify the docs ID / text columns in the `columns` arg, they would face a confusing error (ID / text column not found). This change fixes that problem.